client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/protocol/Codec40.java

```diff
@@ -0,0 +1,38 @@
+package org.infinispan.client.hotrod.impl.protocol;
+
+import java.util.Map;
+
+import org.infinispan.client.hotrod.impl.transport.netty.ByteBufUtil;
+
+import io.netty.buffer.ByteBuf;
+
+/**
+ * @since 14.0
+ */
+public class Codec40 extends Codec31 {
+
+   @Override
+   public HeaderParams writeHeader(ByteBuf buf, HeaderParams params) {
+      return writeHeader(buf, params, HotRodConstants.VERSION_40);
+   }
+
+   @Override
+   protected HeaderParams writeHeader(ByteBuf buf, HeaderParams params, byte version) {
+      HeaderParams headerParams = super.writeHeader(buf, params, version);
+      writeOtherParams(buf, params.otherParams());
+      return headerParams;
+   }
+
+   private void writeOtherParams(ByteBuf buf, Map<String, String> parameters) {
```

Comment:
haven't we decided that it would be `byte[]` instead of `String`?

Reply:
you're right, I'm going to change it